cdnsystem/storedriver/store_mgr.go

```diff
@@ -18,34 +18,48 @@ package storedriver
 
 import (
 	"fmt"
+	"path/filepath"
 	"strings"
 
-	"d7y.io/dragonfly/v2/cdnsystem/config"
 	"d7y.io/dragonfly/v2/cdnsystem/plugins"
+	"d7y.io/dragonfly/v2/pkg/util/fileutils"
+	"github.com/mitchellh/mapstructure"
 )
 
-// StorageBuilder is a function that creates a new storage plugin instant with the giving conf.
-type StorageBuilder func(conf interface{}) (Driver, error)
+// DriverBuilder is a function that creates a new storage driver plugin instant with the giving Config.
+type DriverBuilder func(cfg *Config) (Driver, error)
 
 // Register defines an interface to register a driver with specified name.
 // All drivers should call this function to register itself to the driverFactory.
-func Register(name string, builder StorageBuilder) {
+func Register(name string, builder DriverBuilder) {
 	name = strings.ToLower(name)
 	// plugin builder
-	var f plugins.Builder = func(conf interface{}) (plugin plugins.Plugin, e error) {
-		return NewStore(name, builder, conf)
+	var f = func(conf interface{}) (plugins.Plugin, error) {
+		cfg := &Config{}
+		if err := mapstructure.Decode(conf, cfg); err != nil {
+			return nil, fmt.Errorf("failed to parse config: %v", err)
+		}
+		// prepare the base dir
+		if !filepath.IsAbs(cfg.BaseDir) {
+			return nil, fmt.Errorf("not absolute path: %s", cfg.BaseDir)
+		}
+		if err := fileutils.MkdirAll(cfg.BaseDir); err != nil {
```

Comment:
Use `os.MkdirAll`.

Reply:
Made the default mode encapsulation

Reply:
It is not necessary to use this package.